src/gpuarray_elemwise.c

```diff
@@ -321,20 +323,49 @@ static int check_basic(GpuElemwise *ge, void **args, int flags,
   n = 1;
   for (j = 0; j < nd; j++) {
     p = 0;
+    read_write_arrays_found = 0;
     for (i = 0; i < ge->n; i++) {
       if (is_array(ge->args[i])) {
         v = (GpuArray *)args[i];
-        if (ge->dims[j] != v->dimensions[j]) {
+        if (ge->dims[j] == v->dimensions[j]) {
+          /* We count the number of read-write arrays found until now
+           * that are not broadcasted. */
+          if (is_rw_array(ge->args[i]))
+            ++read_write_arrays_found;
+        } else {
           if (ISCLR(flags, GE_BROADCAST)) {
             return GA_VALUE_ERROR;
           }
           /* GE_BROADCAST is set */
           if (ge->dims[j] == 1) {
```

Comment:
I would add `&& !dim_fixed` here.  This should take care of dimensions that are fixed with the new flag and prevent change if it were to make an output array broadcasted.